File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    for i in range(len(s)) :
        if s[i] == '(' :
            stack.append("(")
        else :
            if stack and stack[-1] == '(' :
                stack.pop()
            else :
                return False
            
    if stack :
        return False
    
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        
        for (char c : s.toCharArray()) {
            if (c == '(') {
                stack.push('(');
            } else {
                if (stack.size() == 0) {
                    answer = false;
                    break;
                } else {
                    if (stack.peek() == '(') {
                        stack.pop();
                    } else {
                        answer = false;
                        break;
                    }
                }
            }
        }
        
        if (stack.size() > 0) {
            answer = false;
        }
        
        return answer;
    }
}